components/org.wso2.carbon.identity.application.authentication.handler.identifier/src/main/java/org/wso2/carbon/identity/application/authentication/handler/identifier/UserResolveExecutor.java

```diff
@@ -0,0 +1,239 @@
+        return resolveUser(tenantAwareUsername, context.getTenantDomain(), context);
+    }
+
+    /**
+     * Rollback logic for the executor. Not implemented.
+     *
+     * @param flowExecutionContext Flow execution context.
+     * @return Always returns null.
+     */
+    @Override
+    public ExecutorResponse rollback(FlowExecutionContext flowExecutionContext) {
+
+        return null;
+    }
+
+    /**
+     * Resolves user attributes and adds them to the flow context.
+     *
+     * @param username     Username to resolve.
+     * @param tenantDomain Tenant domain.
+     * @param context      Flow execution context.
+     */
+    private ExecutorResponse resolveUser(String username, String tenantDomain, FlowExecutionContext context) {
+
+        ExecutorResponse executorResponse;
+        try {
+            UserRealm userRealm = getUserRealm(tenantDomain);
+            if (userRealm == null) {
+                executorResponse = new ExecutorResponse(STATUS_ERROR);
+                executorResponse.setErrorMessage("User realm is not available for tenant: " + tenantDomain);
+                return executorResponse;
+            }
+
+            UserStoreManager userStoreManager = userRealm.getUserStoreManager();
+            String resolvedUsername = resolveQualifiedUsername(username, userStoreManager);
+            Claim[] claims = userStoreManager.getUserClaimValues(resolvedUsername, null);
+            if (claims != null && claims.length > 0) {
+                Map<String, String> claimMap = Arrays.stream(claims)
+                        .filter(c -> c != null && c.getClaimUri() != null)
+                        .collect(Collectors.toMap(Claim::getClaimUri, Claim::getValue));
+                context.getFlowUser().addClaims(claimMap);
+            }
+            executorResponse = new ExecutorResponse(STATUS_COMPLETE);
+
+        } catch (UserStoreException e) {
+            executorResponse = new ExecutorResponse(STATUS_ERROR);
+            executorResponse.setErrorMessage("Error while resolving user '" +
+                    LoggerUtils.getMaskedContent(username) + "' in tenant '" + tenantDomain + "': " + e.getMessage());
+        }
+        return executorResponse;
+    }
+
+    /**
+     * Retrieves the user realm for the given tenant domain.
+     *
+     * @param tenantDomain Tenant domain.
+     * @return UserRealm for the tenant, or null if not available.
+     * @throws UserStoreException If an error occurs while retrieving the user realm.
+     */
+    private UserRealm getUserRealm(String tenantDomain) throws UserStoreException {
+
+        try {
+            RealmService realmService = IdentifierAuthenticatorServiceComponent.getRealmService();
+            int tenantId = realmService.getTenantManager().getTenantId(tenantDomain);
+            return (UserRealm) realmService.getTenantUserRealm(tenantId);
+        } catch (Exception e) {
```

Comment:
Catch the exact exception that's being thrown here. IMO the try block here isn't necessary, as the outer method already handles the exceptions.